app/models/incomeSourceDetails/TaxYear.scala

```diff
@@ -48,6 +48,11 @@ case class TaxYear(startYear: Int, endYear: Int) {
 
   def isBefore(taxYear: TaxYear): Boolean = this.startYear < taxYear.startYear
 
+  def isNextTaxYear(implicit dateService: DateServiceInterface): Boolean = {
+    val currentTaxYearEnd = dateService.getCurrentTaxYearEnd
+    endYear > currentTaxYearEnd && (endYear - currentTaxYearEnd == 1)
+  }
```

Comment:
This method has the Primitive Obsession code smell, and could be implemented more simply using domain concepts with the code `dateService.getCurrentTaxYear.nextYear == this`

Also, passing in Services like `DateService` into value types couples them to those services and reduces the context independence of those value types. So I think this code should be moved off `TaxYear`. Using the above simplification, we could either move the whole expression to the call site, or another alternative might be to split it and add a `nextTaxYear()` method to `DateService` and use something like `dateService.nextTaxYear == xxx` at the call site?

Reply:
done